frontend/src/pages/SearchPage.tsx

```diff
@@ -1,20 +1,30 @@
 import { useSearchRestaurants } from '@/api/RestaurantApi';
+import SearchResultCard from '@/components/SearchResultCard';
+import SearchResultInfo from '@/components/SearchResultInfo';
 import { useParams } from 'react-router-dom';
 
 const SearchPage = () => {
   const { city } = useParams();
-  const { results } = useSearchRestaurants(city);
-  console.log('result :', results);
+  const { results, isLoading } = useSearchRestaurants(city);
+
+  if (isLoading) {
+    return <span>Loading...</span>;
+  }
+
+  if (!results?.data) {
+    return <span>No results found</span>;
+  }
 
   return (
-    <span>
-      Search Page {city}
-      <span>
-        {results?.data.map((restaurant) => (
-          <span key={restaurant._id}>{restaurant.restaurantName}</span>
+    <div className="grid grid-cols-1 lg:grid-cols-[250px-1fr] gap-5">
```

Comment:
<div>

<strong>Bito Code Review Agent Run</strong> <a href=https://github.com/omkardongre/Food-Ordering-App/pull/38#issuecomment-2405784905>#ce9e8f</a><span> - 10/10/2024, 06:36 pm</span>
<br>

<div><small>🔴</small> High importance</div>

<div id="suggestion">
<div id="issue">
<b>Issue:</b> The new layout lacks proper semantic HTML structure, which is crucial for accessibility and SEO.
</div>
<div id="fix">
<b>Fix:</b> Replace generic div elements with semantic HTML elements to provide a clear structure to the page content.
</div>
<details>
<summary>
<b>Code suggestion</b>
</summary>
<div id="code">

```
 @@ -19,9 +19,9 @@ const SearchPage = () => {
 -    <div className="grid grid-cols-1 lg:grid-cols-[250px-1fr] gap-5">
 -      <div id="cuisine-list">insert cuisine here :)</div>
 -      <div id="main-content" className="flex flex-col gap-5">
 +    <main className="grid grid-cols-1 lg:grid-cols-[250px-1fr] gap-5">
 +      <aside id="cuisine-list">insert cuisine here :)</aside>
 +      <section id="main-content" className="flex flex-col gap-5">
          <SearchResultInfo city={city} total={results.pagination.total} />
          {results.data.map((restaurant) => (
            <SearchResultCard key={restaurant._id} restaurant={restaurant} />
          ))}
 -      </div>
 -    </div>
 +      </section>
 +    </main>
```
</div>
</details>
</div>
<br>


</div>

Is this a valid issue, or was it incorrectly flagged by the Agent?
- [ ] it was incorrectly flagged